src/pixelator/pixeldataset.py

```diff
@@ -63,8 +65,30 @@ def read(path: PathType) -> PixelDataset:
     return PixelDataset.from_file(path)
 
 
+def _get_attr_and_index_by_component(attribute, sample_names, datasets):
```

Comment:
I think the reuse of `component` here is a bit confusing between the keyword component we use for not-yet called cells and for the components of the pxl structure as I guess this refers to. 

Can we change the method? Can we standarize how we call things? In the technical documentation pertaining the PXL file, we call them elements.

Reply:
No, this refers to this part:
```
            attr["component"] = pd.Categorical(
                attr["component"].astype(str) + "_" + attr["sample"].astype(str)
            )
            attr.set_index(["component"])
```

That incorporates the sample into the component column (to make it unique), and set's it as the index of the data frame. So I actually think `component` in the name of this function makes sense.